python/lsst/meas/extensions/multiprofit/fit_coadd_multiband.py

```diff
@@ -183,9 +199,401 @@ def __call__(self, source: Mapping[str, Any], *args: Any, **kwargs: Any) -> list
+            cen_x = observation.image.n_cols / 2.0
+        else:
+            cen_x -= centroid_pixel_offset
+        if not np.isfinite(cen_y):
+            # TODO: Add bbox coords or remove
+            cen_y = observation.image.n_rows / 2.0
+        else:
+            cen_y -= centroid_pixel_offset
+
+        # An R_eff larger than the box size is problematic. This should also
+        # stop unreasonable size proposals; a log10 transform isn't enough.
+        # TODO: Try logit for r_eff?
+        size_major = g2.EllipseMajor(g2.Ellipse(sigma_x=sig_x, sigma_y=sig_y, rho=rho)).r_major
+        limits_size = max(5.0 * size_major, 2.0 * np.hypot(x_max - x_min, y_max - y_min))
+        limits_xy = (1e-5, limits_size)
+        params_limits_init = {
+            g2f.CentroidXParameterD: (cen_x, (x_min, x_max)),
+            g2f.CentroidYParameterD: (cen_y, (y_min, y_max)),
+            g2f.ReffXParameterD: (sig_x, limits_xy),
+            g2f.ReffYParameterD: (sig_y, limits_xy),
+            g2f.SigmaXParameterD: (sig_x, limits_xy),
+            g2f.SigmaYParameterD: (sig_y, limits_xy),
+            g2f.RhoParameterD: (rho, None),
+            # TODO: get guess from configs?
+            g2f.SersicMixComponentIndexParameterD: (1.0, None),
+        }
+
+        idx_obs = 0
+        for param in self.params_init:
+            if param.linear:
+                value_init = fluxes_init[idx_obs]
+                limits_new = fluxes_limits[idx_obs]
+                idx_obs += 1
+                if idx_obs == n_observations:
+                    idx_obs = 0
+            else:
+                type_param = type(param)
+                value_init, limits_new = params_limits_init.get(type_param, (values_init.get(param), None))
+            if limits_new:
+                param.limits = g2f.LimitsD(limits_new[0], limits_new[1])
+            if value_init is not None:
+                param.value = value_init
+
+        priors_shape_mag = self.priors_shape_mag
+        has_priors_mag = len(priors_shape_mag) > 0
+        if has_priors_mag:
+            mag_total = u.nJy.to(u.ABmag, np.nansum(fluxes_init))
+
+        # TODO: Add centroid prior
+        priors_gauss, priors_shape = self.get_priors_type(model)
+        for prior in priors_shape:
+            if has_priors_mag and ((prior_adjustments := priors_shape_mag.get(prior)) is not None):
+                mag_dep_prior, prior_shape_new = prior_adjustments
+                prior_size_new = prior_shape_new.prior_size
+                prior.prior_size.mean_parameter.value = prior_size_new.mean_parameter.value * 10**(
+                    # the size-apparent mag relation probably flattens
+                    # for very bright/faint objects - maybe not so
+                    # sharply, but this should work fine.
+                    mag_dep_prior.slope_median_per_mag*np.clip(
+                        mag_total - mag_dep_prior.intercept_mag, -12.5, 12.5,
+                    )
+                )
```

Comment:
I would move the comments to before the statement, not the middle of it. Same for the value that is set below. If you feel like that specific line needs a comment in place, then I would define a variable and set it before setting `prior.prior_size.mean_parameter.value`.